### PR TITLE
Splunk Audit

### PR DESCRIPTION
**Pull request by TGPSKI**
This PR adds Splunk Audit capabilities to gabi's query handler.

* `pkg/env/splunk`: follows the dbenv pattern in sourcing variables from env / provided by k8s
* `pkg/audit/audit.go`: adds `SplunkAudit` and supporting types and structs. `SplunkAudit.Write()` handles constructing the Splunk payload and POSTing it to the collector endpoint
* `pkg/handlers/query.go`: updated to log to both stdout and Splunk. If Splunk logging fails, the request is not executed. The return type has been updated to return the result array and an error field as well.
* `openshift/gabi.template.yaml`: added environment variables to the pod template to support using SplunkAudit

Example event payload in Splunk:

image

Local server hitting local db:

image

### CODE REVIEW

File: pkg/audit/audit.go [hunk 1]
```diff
@@ -27,8 +35,92 @@ func (d *LoggingAudit) Write(q *QueryData) error {
...
+	Token      string
+	Endpoint   string
+	Host       string
+	Pod        string
+	Namespace  string
+	Source     string
+	Sourcetype string
+}
+
+func (d *SplunkAudit) Init(se *splunk.Splunkenv) {
```
Comment (TGPSKI) [inline pkg/audit/audit.go:70]: Passing these variables around like this feels a bit awkward. I am thinking on how to simplify and I'm open to any ideas.

**Review by BumbleFeng — COMMENTED** [review]
lgtm

File: pkg/audit/audit.go (hunk 1 above)
Comment (BumbleFeng) [reply pkg/audit/audit.go:70]: I think you can just define 
```
type SplunkAudit struct {
  env Splunkenv
}
```

**Review by BumbleFeng — COMMENTED** [review]
lgtm

File: pkg/audit/audit.go (hunk 1 above)
Comment (BumbleFeng) [reply pkg/audit/audit.go:70]: I think you can just define 
```
type SplunkAudit struct {
  env Splunkenv
}
```

**Pull request merged**